jobrunner/log_utils.py

```diff
@@ -18,46 +17,36 @@
 DEFAULT_FORMAT = "{asctime} {message} {tags}"
 
 
-def formatting_filter(record):
-    """Ensure various record attribute are always available for formatting."""
-
-    # ensure this are always available for static formatting
-    record.action = ""
-
-    tags = {}
-    ctx = set_log_context.current_context
-    job = getattr(record, "job", None) or ctx.get("job")
-    req = getattr(record, "job_request", None) or ctx.get("job_request")
-
-    status_code = getattr(record, "status_code", None)
-    if status_code:
-        tags["status"] = record.status_code
-
-    if job:
-        # preserve short action for local run formatting
-        record.action = job.action + ": "
-        if "status" not in tags and job.status_code:
-            tags["status"] = job.status_code
-        tags["project"] = job.project
-        tags["action"] = job.action
-        tags["id"] = job.id
-
-    if req:
-        tags["req"] = req.id
-
-    record.tags = " ".join(f"{k}={v}" for k, v in tags.items())
-
-    return True
-
-
 def configure_logging(fmt=DEFAULT_FORMAT, stream=None, status_codes_to_ignore=None):
     formatter = JobRunnerFormatter(fmt, style="{")
     handler = logging.StreamHandler(stream=stream)
     handler.setFormatter(formatter)
     if status_codes_to_ignore:
         handler.addFilter(IgnoreStatusCodes(status_codes_to_ignore))
     handler.addFilter(formatting_filter)
-    logging.basicConfig(level=os.environ.get("LOGLEVEL", "INFO"), handlers=[handler])
+
+    log_level = os.environ.get("LOGLEVEL", "INFO")
+    handlers = [handler]
+
+    # Support a separate log file at level DEBUG, while leaving the default
+    # logs untouched. DEBUG logging can be extremely noisy and so we want a way
+    # to capture these that doesn't pollute the primary logs.
+    debug_log_file = os.environ.get("DEBUG_LOG_FILE")
+    if debug_log_file:
+        debug_handler = logging.FileHandler(debug_log_file, encoding="utf-8")
```

Comment:
Maybe add a simple limits to avoid accidentally leaving this on and filling up disk

I've done this kind of thing before: https://github.com/canonical-ols/talisker/blob/master/talisker/logs.py#L156